src/doc/reference.md

```diff
@@ -3616,38 +3617,26 @@ x = bo(5,7);
 
 ### Closure types
 
-```{.ebnf .notation}
-closure_type := [ 'unsafe' ] [ '<' lifetime-list '>' ] '|' arg-list '|'
-                [ ':' bound-list ] [ '->' type ]
-lifetime-list := lifetime | lifetime ',' lifetime-list
-arg-list := ident ':' type | ident ':' type ',' arg-list
-bound-list := bound | bound '+' bound-list
-bound := path | lifetime
-```
-
-The type of a closure mapping an input of type `A` to an output of type `B` is
-`|A| -> B`. A closure with no arguments or return values has type `||`.
-
-An example of creating and calling a closure:
+A [lambda expression](#lambda-expressions) produces a closure value with
+a unique, anonymous type that cannot be written out.
 
-```rust
-let captured_var = 10;
+Depending on the requirements of the closure, its type implements one or
+more of the closure traits:
 
-let closure_no_args = || println!("captured_var={}", captured_var);
+* `FnOnce`
+  : The closure can be called at least once. A closure called as `FnOnce`
+    can move out values from its environment.
```

Comment:
No. All FnMut closures impl FnOnce too. They can be called at least once since they impl FnOnce. It's a bit tricky how to express this but I was generally happy with how I presented the cascade of traits here.